### PR TITLE
Added functionality to count rate of center

### PR DESCRIPTION
**Pull request by NadiiaNanivska**
Added a new endpoint in the CenterController to retrieve all clubs by center ID.
Updated the ClubRepository to include a method for retrieving all clubs by center ID.
Expanded the CenterService with method to retrieve list of all clubs.
Implemented method in the CenterServiceImpl to fetch clubs by center ID.

### CODE REVIEW

File: src/main/java/com/softserve/teachua/controller/CenterController.java [hunk 1]
```diff
@@ -80,13 +80,17 @@ public Page<CenterResponse> getCentersByUserId(@PathVariable Long id,
      * @return new {@code Page<ClubResponse>}.
      */
     @GetMapping("centers/clubs/{id}")
-    public Page<ClubResponse> getCenterClubsByCenterId(@PathVariable Long id,
-                                                       @RequestParam int size, // size of pagination
-                                                       @PageableDefault(sort = "id")
-                                                       Pageable pageable) {
-        return centerService.getCenterClubsByCenterId(id, pageable);
+    public Object getCenterClubsByCenterId(@PathVariable Long id,
```
Comment (BNazar00) [inline src/main/java/com/softserve/teachua/controller/CenterController.java]: Consider using more specific return types. The method's return type is currently defined as `Object`, I'd not recommend doing so.

**Review by BNazar00 — CHANGES_REQUESTED** [review]
First of all, please ensure that you have familiarized yourself with the pull request template, which you seem to have deleted. It is crucial to retain the 'Checklist' section and indicate the items that are necessary.

Additionally, it would be immensely helpful if you could provide a brief description of your pull request. If you have resolved an issue, mention it and include a link to the corresponding issue for reference (if such is) or shortly describe what you've fixed.

Furthermore, I kindly request that you address the comment from SonarBot on your pull request. It is essential to make an effort to avoid introducing any new issues. Presently, there is one 'Code Smell' that requires attention. The SonarBot comment provides further details regarding this matter.

File: src/main/java/com/softserve/teachua/controller/CenterController.java (hunk 1 above)
Comment (NadiiaNanivska) [reply src/main/java/com/softserve/teachua/controller/CenterController.java]: Done

File: src/main/java/com/softserve/teachua/controller/CenterController.java [hunk 2]
```diff
@@ -87,6 +87,11 @@ public Page<ClubResponse> getCenterClubsByCenterId(@PathVariable Long id,
         return centerService.getCenterClubsByCenterId(id, pageable);
     }
 
+    @GetMapping("centers/allclubs/{id}")
+    public Object getAllCenterClubsByCenterId(@PathVariable Long id) {
```
Comment (BNazar00) [inline src/main/java/com/softserve/teachua/controller/CenterController.java]: Regarding the return type, please see my previous comment for the suggestion of using a more specific return type.

Comment (BNazar00) [inline src/main/java/com/softserve/teachua/controller/CenterController.java]: I'd also highly recommend following the project's API routes convention. If you have chosen to associate this functionality with the `CenterController` and there is a similar route, consider using the parameter, like
 ```suggestion
     @GetMapping("/centers/clubs", params = "centerId")
     public Object getAllCenterClubsByCenterId(@RequestParam("centerId) Long id) {
```


I'd relate such functionality to `ClubController`, though.

**Review by BNazar00 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**